src/borg/archiver.py

```diff
@@ -1129,6 +1129,64 @@ def do_with_lock(self, args, repository):
     def do_debug_info(self, args):
         """display system information for debugging / bug reports"""
         print(sysinfo())
+
+    def do_debug_speed(self, args):
+        components = collections.OrderedDict()
+
+        def component(func):
+            @functools.wraps(func)
+            def wrapper(size):
+                td = 0.0
+                for i in range(args.trials):
+                    print('Measuring', func.__name__, '(%d/%d)' % (i + 1, args.trials), end='\r')
+                    t0 = time.perf_counter()
+                    func(size)
+                    t1 = time.perf_counter()
+                    td += t1 - t0
+                print(func.__name__, 'speed: %.1f MB/s' % (args.trials * size / td / 1000**2))
+            components[func.__name__] = wrapper
+            return wrapper
+
+        @component
+        def chunker(size):
+            from .chunker import Chunker
+            from .constants import CHUNKER_PARAMS
+            import io
+            chunker = Chunker(0, *CHUNKER_PARAMS)
+            for _ in chunker.chunkify(io.BytesIO(bytes(size))):
+               pass
+
+        @component
+        def hmac_sha256(size):
+            from .crypto import hmac_sha256
+            hmac_sha256(bytes(32), bytes(size))
+
+        @component
+        def blake2b(size):
+            from .crypto import blake2b_256
+            blake2b_256(bytes(32), bytes(size))
+
+        @component
+        def crc32_zlib(size):
+            from zlib import crc32
+            crc32(bytes(size))
```

Comment:
No, there is a difference though due to the way large memory areas are allocated (mapping the same physical page many times), meaning that compared to real data laid out across a memory access will be faster. This reduces throughput (of all implementations), but affects the faster one stronger, of course.

However, since Borg works on chunks that are just processed before they're likely in some cache, so solely looking at the streaming case is probably not quite right, too. Like I wrote above - it's a crude tool, more for telling the ballpark.